src/main/clojure/clara/rules/compiler.clj

```diff
@@ -1263,105 +1439,70 @@
    id :- sc/Int
    is-root :- sc/Bool
    children :- [sc/Any]
-   compile-expr ; Function to do the evaluation.
+   id-key->expr :- {(schema/tuple sc/Int sc/Keyword) IFn}
    new-bindings :- #{sc/Keyword}]
 
   (let [{:keys [condition production query join-bindings]} beta-node
 
         condition (if (symbol? condition)
                     (.loadClass (clojure.lang.RT/makeClassLoader) (name condition))
-                    condition)]
+                    condition)
+
+        retrieve-fn (fn [id field] (get id-key->expr [id field]))]
```

Comment:
Recommended rename: `compiled-expr-fn`. This is too generic and confusing to read later on.